helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -807,7 +807,9 @@ private RollingUpdateOp rollingUpdateDeploy(final ZooKeeperClient client,
                                                 deploymentGroup.getName());
 
     try {
-      return opFactory.nextTask(getDeployOperations(client, host, deployment, Job.EMPTY_TOKEN));
+      final String token = deploymentGroup.getRolloutOptions().getToken() == null ?
```

Comment:
If you want, you could just do:

`final String token = firstNonNull(deploymentGroup.getRolloutOptions.getToken(), Job.EMPTY_TOKEN);`